src/interner.rs

```diff
@@ -82,12 +82,12 @@ impl<S, B, H> Clone for StringInterner<S, B, H>
 where
     S: Symbol,
     B: Backend<S> + Clone,
-    H: BuildHasher + Default,
+    H: BuildHasher + Clone,
 {
     fn clone(&self) -> Self {
         Self {
             dedup: self.dedup.clone(),
-            hasher: Default::default(),
+            hasher: self.hasher.clone(),
```

Comment:
I believe the problem is that there is actually state in the `hasher` so creating a new instance of it means that the `hasher` instances built from it will produce different hash values for the same input. 

From the rust docs: https://doc.rust-lang.org/std/collections/hash_map/struct.RandomState.html

> A particular instance RandomState will create the same instances of Hasher, but the hashers created by two different RandomState instances are unlikely to produce the same result for the same values.

So in this case calling `default` means the clone is using a different `RandomState` instance and thus has different hash values